plugins/aws_workspaces/icon_aws_workspaces/actions/create_workspace/action.py

```diff
@@ -14,86 +14,84 @@ def __init__(self):
             output=CreateWorkspaceOutput(),
         )
 
-    def run(self, params={}):
-        directory_id = params.get(Input.DIRECTORY_ID)
-        username = params.get(Input.USERNAME)
-        bundle_id = params.get(Input.BUNDLE_ID)
-        volume_encryption_key = params.get(Input.VOLUME_ENCRYPTION_KEY)
-        user_volume_encryption_enabled = params.get(Input.USER_VOLUME_ENCRYPTION_ENABLED)
-        root_volume_encryption_enabled = params.get(Input.ROOT_VOLUME_ENCRYPTION_ENABLED)
-        workspace_properties = params.get(Input.WORKSPACE_PROPERTIES)
-        tags = params.get(Input.TAGS)
-        result = {}
+    def build_payload(self, params: dict) -> dict:
 
         payload = {
-            "DirectoryId": directory_id,
-            "UserName": username,
-            "BundleId": bundle_id,
-            "Tags": tags,
+            "DirectoryId": params.get(Input.DIRECTORY_ID),
+            "UserName": params.get(Input.USERNAME),
+            "BundleId": params.get(Input.BUNDLE_ID),
+            "Tags": params.get(Input.TAGS),
             "WorkspaceProperties": {
-                "ComputeTypeName": workspace_properties["compute_type_name"],
-                "RootVolumeSizeGib": workspace_properties["root_volume_size"],
-                "RunningMode": workspace_properties["running_mode"],
-                "RunningModeAutoStopTimeoutInMinutes": workspace_properties["running_mode_auto_stop_time_out"],
-                "UserVolumeSizeGib": workspace_properties["user_volume_size"],
+                "ComputeTypeName": params.get(Input.WORKSPACE_PROPERTIES)["compute_type_name"],
```

Comment:
It would have thrown an error on the old version as well, but workspace properties aren't required so this could error if not supplied? Are we bett careting a workspace_properties like before (line 24), and defaulting to `{}`, then accessing each of these keys here using .get()?